### PR TITLE
Local executor

### PR DESCRIPTION
**Pull request by intrepiditee**
1. Adds a local executor that performs the equivalent of '/update' (takes an absolute import name and produces data files). See local_executor.py.
2. executor.py renamed to import_executor.py
3. The executor is now a class (ImportExecutor)
4. The executor no longer calls chdir
5. Introduces ExecutionResult to report back to the caller which imports executed successfully in case some of them fail

Thanks for reviewing.

### CODE REVIEW

File: import-automation/executor/local_executor.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Local import executor. Run '. run_local_executor.sh --help' for usage.
+
+The local executor downloads the master branch of a repository and produces
```
Comment (jeffreyoldham) [inline import-automation/executor/local_executor.py:18]: Is it master xor "master"?  The latter indicates a particular name.  The former means the code asks the repository for its main branch.

File: import-automation/executor/local_executor.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+the data files of an import specified by its absolute import name of the form
+<path to the directory containing the manifest>:<import name>.
+
+username and access_token are used for authentication with GitHub to access
+private repositories and get higher rate limits. They need to be both absent or
+provided. See
+https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token
+for how to create access tokens.
+
+owner_username and repo_name uniquely identify a repository. In
```
Comment (jeffreyoldham) [inline import-automation/executor/local_executor.py:28]:
```suggestion
owner_username and repo_name uniquely identify a repository. E.g., in
```

Comment (intrepiditee) [reply import-automation/executor/local_executor.py:28]: fixed

File: import-automation/executor/local_executor.py (hunk 1 above)
Comment (intrepiditee) [reply import-automation/executor/local_executor.py:18]: The main branch

Comment (intrepiditee) [reply import-automation/executor/local_executor.py:18]: Changed to 'the main branch of ...'

**Pull request merged**